src/main/java/frc/robot/commands/AutoVisionCommand.java

```diff
@@ -0,0 +1,78 @@
+import frc.robot.subsystems.conveyor.Conveyor;
+import frc.robot.subsystems.drive.DriveSubsystem;
+import frc.robot.subsystems.elevator.Elevator;
+import frc.robot.subsystems.intake.Intake;
+import frc.robot.subsystems.intakeLimelight.IntakeLimelight;
+import frc.robot.subsystems.lights.Lights;
+import frc.robot.subsystems.shooter.Shooter;
+import java.util.List;
+import java.util.Optional;
+
+public class AutoVisionCommand extends SequentialCommandGroup {
+  public Optional<PathPlannerPath> path = Optional.empty();
+  public Optional<Pair<Pose2d, Pose2d>> poses = Optional.empty();
+  public AutoVisionCommand(
+      DriveSubsystem drive,
+      Intake intake,
+      Conveyor conveyor,
+      Elevator elevator,
+      Shooter shooter,
+      Lights lights,
+      IntakeLimelight intakeLimelight) {
+    addRequirements(drive, intake, conveyor, elevator);
+    addCommands(
+        new SequentialCommandGroup(
+            new IntakeSequence(intake, elevator, conveyor, shooter, lights),
+            new InstantCommand(
+                () -> {
+                  intakeLimelight.setLimelightValues(
+                      limelightLedMode.OFF,
+                      limelightCamMode.VISION_PROCESSING,
+                      limelightPipeline.NOTE_PIPELINE);
+                  Optional<frc.robot.subsystems.intakeLimelight.IntakeLimelight.NoteDetection>
+                      latestNoteDetectionOptional = Optional.empty();
+                  latestNoteDetectionOptional = intakeLimelight.getNotePos(true);
+                  if (latestNoteDetectionOptional.isPresent()) {
+                    var latestNoteDetection = latestNoteDetectionOptional.get();
+                    final double adjustmentMeters = Units.inchesToMeters(6.0);
+                    Translation2d poseAtDetectionToNote =
+                        new Translation2d(
+                                latestNoteDetection.distanceMeters - adjustmentMeters, 0.0)
+                            .rotateBy(Rotation2d.fromDegrees(latestNoteDetection.yawAngleDeg));
+                    Pose2d robotPoseAtDetection =
+                        drive.getPastBufferedPose(latestNoteDetection.latencySec);
+                    Pose2d curPose = drive.getPose();
+                    Pose2d goalPose =
+                        robotPoseAtDetection.plus(
+                            new Transform2d(
+                                poseAtDetectionToNote,
+                                Rotation2d.fromDegrees(latestNoteDetection.yawAngleDeg)));
+                    List<Translation2d> bezierPoints = PathPlannerPath.bezierFromPoses(
+                      curPose,
+                      goalPose
+                    );
+                    poses = Optional.of(new Pair<Pose2d, Pose2d>(curPose, goalPose));
+                    path = Optional.of(new PathPlannerPath(bezierPoints, new PathConstraints(5.0, 6.0, 2*Math.PI, 4*Math.PI), new GoalEndState(2.5, Rotation2d.fromDegrees(latestNoteDetection.yawAngleDeg))));
```

Comment:
can you make these variables in constants or cal

Reply:
later